### PR TITLE
DEVPROD-11579 Update deprecated go driver code to non-deprecated

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-11579

### Description
This is a small bump in the syntax we use for the go driver. The original ticket is dogfooding the beta of the driver, so this was a byproduct of me updating it- so that whoever updates to the real go driver v2 doesn't have to do these changes.

### Testing
Existing testing

### CODE REVIEW

File: db/db_utils.go
Comment (ZackarySantana) [inline db/db_utils.go:33]: I didn't include this in the comment but we could add a TODO comment pointing to [DEVPROD-11824](https://jira.mongodb.org/browse/DEVPROD-11824) and make a comment there about this function (I'm planning on picking it up and I know about it, the TODO/comment is just if we want to provide support if someone else picks it up).

The reason we need it is because in go driver v2, setting the timeout is now handled by cancelling a context (context.withtimeout) rather than setting an option- so our db queries need a context. So technically dead code right now so I could remove it if the reviewer wants!

**Comment by ZackarySantana** [comment]
The failing test is just the agent version. I'm going to update it once i get approval to avoid merge conflicts with it

File: db/db_utils.go
Comment (ablack12) [inline db/db_utils.go:62]: can we add a comment here since this is a newly added public function?

File: db/db_utils.go
Comment (ablack12) [reply db/db_utils.go:33]: I think it's worth still adding a TODO since the ticket isn't assigned to you, but if you want to assign it that works too!

also nit: "passed-in context"

File: db/db_utils.go
Comment (ablack12) [inline db/db_utils.go:69]: should we error "context session is not defined" to distinguish this error from the above function?

File: db/db_utils.go
Comment (ablack12) [reply db/db_utils.go:33]: Also will we still need GetSession, or will we remove this as part of DEVPROD-11824 as well?

**Review by ablack12 — APPROVED** [review]

File: db/db_utils.go
Comment (ZackarySantana) [reply db/db_utils.go:33]: I think we might be able to remove it. I'll add a TODO comment!

**Pull request merged**
